### ISSUE TITLE
clean_text() confusion

### ISSUE THREAD

**Issue by BoboTiG** [timovn/blogotext#50]
I installed BlogoText from the dev branch and when editing/creating a post there are many slashes added to the abstract.
So, I looked for the function that add slashes and find that it is currently `clean_txt()` from `inc/conv.php`. I also found that many times you call `stripslashes(clean_txt())`, it seems confusing (I cleanup a little):
```shell
inc/conv.php:
htmlspecialchars(stripslashes(clean_txt(...)))

inc/fich.php:	
addslashes(clean_txt(htmlspecialchars(...)))
clean_txt(htmlspecialchars(...))
clean_txt(htmlspecialchars(...))
clean_txt(htmlspecialchars(...))
clean_txt(htmlspecialchars(...))
clean_txt(htmlspecialchars(...))
addslashes(clean_txt(htmlspecialchars(...)))
addslashes(clean_txt(htmlspecialchars(...)))
clean_txt(htmlspecialchars(...))
clean_txt(...)
clean_txt(trim(htmlspecialchars(...)))  # trim() is useless

inc/sqli.php:	
clean_txt(...);
stripslashes(clean_txt(...))
stripslashes(clean_txt(...))
htmlspecialchars(clean_txt(...))
stripslashes(clean_txt(...))
htmlspecialchars(stripslashes(clean_txt(...)))

inc/imgs.php:
stripslashes(clean_txt(...))
stripslashes(clean_txt(...))
```

The use of `clean_txt` with other functions is weird, sometimes repetitive or one cancels the other. Are you working on something unified and "less error prone"? Or I can just send a patch for the abstract issue.

**Comment by timovn** [comment]
I know that part is a bit messy, even legacy from the text-based Blogotext. The addshlashes normally is called when the server does not add them by default. Then comes the stripslashes part.

I'll have a look at this.

**Comment by BoboTiG** [comment]
So, for the moment, to fix the abstract issue:
```diff
diff --git a/inc/sqli.php b/inc/sqli.php
--- a/inc/sqli.php
+++ b/inc/sqli.php
@@ -257,7 +257,7 @@ function init_post_article() { //no $mode : it's always admin.
                'bt_id'                         => $id,
                'bt_date'                       => $date,
                'bt_title'                      => protect($_POST['titre']),
-               'bt_abstract'           => (empty($_POST['chapo']) ? '' : clean_txt($_POST['chapo'])),
+               'bt_abstract'           => (empty($_POST['chapo']) ? '' : stripslashes(clean_txt($_POST['chapo']))),
                'bt_notes'                      => protect($_POST['notes']),
                'bt_content'            => $formated_contenu,
                'bt_wiki_content'       => stripslashes(clean_txt($_POST['contenu'])),
```

Do you want I send a PR?

**Comment by timovn** [comment]
I see the problem.
No need for PR, I fixed the whole thing by changing the `clean_txt()` function. It will be pushed in a few day with other issues (can't push now). 

Thanks for the help!

**Comment by BoboTiG** [comment]
Great! :)

**Comment by timovn** [comment]
Should be fixed with 50200fb
